staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -126,12 +127,11 @@ func ConvertGroupVersionIntoToDiscovery(list []metav1.APIResource) ([]apidiscove
 		parent, exists = parentResources[split[0]]
 		if !exists {
 			// If a subresource exists without a parent, create a parent
-			apiResourceList = append(apiResourceList, apidiscoveryv2beta1.APIResourceDiscovery{
+			parent = &apidiscoveryv2beta1.APIResourceDiscovery{
 				Resource: split[0],
 				Scope:    scope,
-			})
-			parentResources[split[0]] = &apiResourceList[len(apiResourceList)-1]
-			parent = &apiResourceList[len(apiResourceList)-1]
+			}
+			apiResourceList = append(apiResourceList, parent)
 			parentResources[split[0]] = parent
```

Comment:
I dont think it was necessary to change the whole array to pointers. Rather than assigning parent to the address of an index you can assign it to the address of a variable similar to something like:

```go
parent = &apidiscoveryv2beta1.APIResourceDiscovery{...}
apiResourceList = append(apiResourceList, *parent)
```

Reply:
If we append to apiResourceList by value rather than by pointer, we cannot directly update the parent resource (in the subresources loop) without searching for the parent object in the array. That seems inefficient.

Reply:
can we just store the index then?

Reply:
Yep, updated